### PR TITLE
feat: Add TagColored to component temp 

### LINKED ISSUE #8676: Add TagColored to component-temp
- Add TagColored to component-temp folder based on temp design at https://www.figma.com/file/PsKqriAOpJDQpaHt373D45/Off-Ramp?type=design&node-id=1%3A2&mode=design&t=pPqrfRmG4eNsCDPZ-1
- TagColor: Default, Success, Info, Danger, Warning

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Added TagColored to component temp folder, based on https://www.figma.com/file/PsKqriAOpJDQpaHt373D45/Off-Ramp?type=design&node-id=1%3A2&mode=design&t=pPqrfRmG4eNsCDPZ-1

## **Related issues**

Fixes: #8676 

## **Manual testing steps**

1. Change pointer in `index.js` to Storybook to open Storybook
2. Click on Components Temp
3. Click on TagColored

### **After**
https://github.com/MetaMask/metamask-mobile/assets/14355083/12c727d0-474c-4638-aae5-ac551b4f81d6

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by wachunei** [comment]
LGTM!

**Review by wachunei — DISMISSED** [review]

File: .storybook/main.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   stories: [
     '../app/component-library/components/**/*.stories.?(ts|tsx|js|jsx)',
+    '../app/component-library/components-temp/TagColored/**/*.stories.?(ts|tsx|js|jsx)',
```
Comment (garrettbear) [inline .storybook/main.js:4]: I won't block you here, but I probably would have made this more universal by not using `TagColored` and utilizing the globstar pathname 
```suggestion
    '../app/component-library/components-temp/**/*.stories.?(ts|tsx|js|jsx)',
```

Comment (garrettbear) [reply .storybook/main.js:4]: Don't worry about this as much. I know this is just a temp thing

**Review by garrettbear — APPROVED** [review]
LGTM. My mobile repo isn't running locally but I saw Pedro's screenshot and it looks good.

**Review by wachunei — APPROVED** [review]

**Pull request merged**